.github/workflows/push-build.yaml

```diff
@@ -0,0 +1,58 @@
+name: Push Image Build
+on:
+  push:
+    branches:
+      - master
+      - release-*
+    tags:
+      - v*
+
+defaults:
+  run:
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+jobs:
+  push-image-to-container-registry:
+    runs-on: ubuntu-18.04
+    if: github.repository == 'rook/rook'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+       # docker/setup-qemu action installs QEMU static binaries, which are used to run builders for architectures other than the host.
+      - name: set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: all
+
+      - name: log in to container registry
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_USR }}
+          aws-secret-access-key: ${{ secrets.AWS_PSW }}
+          aws-region: us-east-1
+
+      # creating custom env var
+      - name: set env
+        run: |
+          echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
+          echo "GITHUB_REF"=${GITHUB_REF} >> $GITHUB_ENV
+
+      - name: build and release
+        env:
+          GIT_API_TOKEN: ${{ secrets.GIT_API_TOKEN }}
+          BRANCH_NAME: ${{ env.BRANCH_NAME }}
+          AWS_USR: ${{ secrets.AWS_USR }}
+          AWS_PSW: ${{ secrets.AWS_PSW }}
+          GITHUB_REF: $ {{ env.GITHUB_REF }}
+        run: |
+          tests/scripts/build-release.sh
```

Comment:
Do we need to backport all these build scripts as well?

Reply:
added this script and gen_release_notes, not adding GitHub action release note script as it will make it bigger change(and more work :grin: )